Source/DafnyCore/Resolver/PreType/PreTypeResolve.cs

```diff
@@ -404,10 +404,11 @@ public enum Type2PreTypeOption { GoodForInference, GoodForPrinting, GoodForBoth
     private class PreTypeInferenceModuleState {
       public readonly ISet<Declaration> StillNeedsPreTypeSignature;
       public readonly Stack<Declaration> InFirstPhase = new Stack<Declaration>();
-      public readonly Dictionary<string, TopLevelDecl> PreTypeBuiltins = new();
```

Comment:
Why these changes? Do we need to be able to use different PreTypeBuiltins Dictionnaries?

Reply:
It's so that `PreTypeBuiltins` can be defined in `SystemModuleManager` instead. In this way, all modules will use the same `PreTypeBuiltins`, rather than getting a different one for each module. And this, in turn, is needed in order for the pre-type resolver to properly understand, for example, that there's just one `int` type.